CMakeLists.txt

```diff
@@ -16,58 +16,70 @@ if (BUILD_SHARED_LIBS)
-                                  PATH_VARS INCLUDE_INSTALL_DIR
-                                  NO_CHECK_REQUIRED_COMPONENTS_MACRO)
+      # Export the targets  (build tree)
+      export(EXPORT dlfcn-win32-targets
+             FILE "${CMAKE_CURRENT_BINARY_DIR}/dlfcn-win32-targets.cmake"
+             NAMESPACE dlfcn-win32::
+      )
 
-    # Install the targets (install)
-    install(EXPORT dlfcn-win32-targets
-            FILE dlfcn-win32-targets.cmake
-            NAMESPACE dlfcn-win32::
-            DESTINATION ${CMAKE_CONF_INSTALL_DIR})
+      # Write the CMake config file
+      set(CMAKE_CONF_INSTALL_DIR share/dlfcn-win32)
+      set(INCLUDE_INSTALL_DIR include)
+      include(CMakePackageConfigHelpers)
+      configure_package_config_file(dlfcn-win32-config.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/dlfcn-win32-config.cmake
+                                    INSTALL_DESTINATION ${CMAKE_CONF_INSTALL_DIR}
+                                    PATH_VARS INCLUDE_INSTALL_DIR
+                                    NO_CHECK_REQUIRED_COMPONENTS_MACRO)
 
-    # Install the CMake config file
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/dlfcn-win32-config.cmake
-            DESTINATION ${CMAKE_CONF_INSTALL_DIR})
+      # Install the targets (install)
+      install(EXPORT dlfcn-win32-targets
+              FILE dlfcn-win32-targets.cmake
+              NAMESPACE dlfcn-win32::
+              DESTINATION ${CMAKE_CONF_INSTALL_DIR})
+
+      # Install the CMake config file
+      install(FILES ${CMAKE_CURRENT_BINARY_DIR}/dlfcn-win32-config.cmake
+              DESTINATION ${CMAKE_CONF_INSTALL_DIR})
+  endif()
 endif()
 
 if (BUILD_TESTS)
   enable_testing()
-  add_library(testdll SHARED testdll.c)
-  set_target_properties(testdll PROPERTIES PREFIX "")
-  add_library(testdll2 SHARED testdll2.c)
-  set_target_properties(testdll2 PROPERTIES PREFIX "")
-  target_link_libraries(testdll2 dl)
-  add_library(testdll3 SHARED testdll3.c)
-  set_target_properties(testdll3 PROPERTIES PREFIX "")
-  add_executable(t_dlfcn test.c)
-  target_link_libraries(t_dlfcn dl)
-  add_test (NAME t_dlfcn COMMAND t_dlfcn)
+  add_definitions(-DLIB_SUFFIX="${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  if(WIN32)
+    add_library(testdll SHARED testdll.c)
+    set_target_properties(testdll PROPERTIES PREFIX "")
+    add_library(testdll2 SHARED testdll2.c)
+    set_target_properties(testdll2 PROPERTIES PREFIX "")
+    target_link_libraries(testdll2 dl)
+    add_library(testdll3 SHARED testdll3.c)
+    set_target_properties(testdll3 PROPERTIES PREFIX "")
+    add_executable(t_dlfcn test.c)
+    target_link_libraries(t_dlfcn dl)
+    add_test (NAME t_dlfcn COMMAND t_dlfcn)
+    include_directories(.)
+  else()
+    add_executable(test-dladdr-pie test-dladdr.c)
+    target_compile_options(test-dladdr-pie PRIVATE -fpie)
+    target_link_libraries(test-dladdr-pie dl)
+  endif()
+  add_executable(test-dladdr test-dladdr.c)
+  target_link_libraries(test-dladdr dl)
```

Comment:
It looks like in this file are lot of whitespace changes and such thing is hard to review. Please avoid such big whitespace changes if it is possible.

Reply:
Did you saw the related git commit message: 
```
    Add test-dladdr test application
    
    The application is only build on UNIX like OS to get an overview what
    is implemented.
```
For building test-dladdr on non Windows platform all other targets needs to be excluded, which is implemented in that patch.

Reply:
Now I saw, but I do not think that somebody would expect that library with explicit win32 substring in name is going to be used on non-windows platforms. Looks like that it is not needed.

Reply:
Assuming you have another Unix-like OS system, how would you get detailed information about its dladdr implementation unless you have something like test-dladdr that could be compiled and run on that platform ?

Reply:
I see. But I do not think it should be part of main CMakeLists.txt. Basically such utility is not needed for end users who compile project and therefore it also should not be enabled by default. Also it is not needed for compiling project at all. It is useful for developers and that is why I think it should be separated.

Reply:
Removed unix os support from this pull request, which looks better to be in a separate pull request.

Reply:
See https://github.com/dlfcn-win32/dlfcn-win32/pull/79. After it has been merged I will update this pr.